src/webrtc/call.ts

```diff
@@ -1713,17 +1700,34 @@ function setTracksEnabled(tracks: Array<MediaStreamTrack>, enabled: boolean) {
     }
 }
 
-function getUserMediaVideoContraints(callType: CallType) {
+function getUserMediaContraints(type: ConstraintsType) {
     const isWebkit = !!navigator.webkitGetUserMedia;
 
-    switch (callType) {
-        case CallType.Voice:
+    switch (type) {
+        case ConstraintsType.Audio: {
             return {
                 audio: {
                     deviceId: audioInput ? {ideal: audioInput} : undefined,
-                }, video: false,
+                },
+                video: false,
             };
-        case CallType.Video:
+        }
+        case ConstraintsType.Video: {
```

Comment:
Doesn't look like this is used - would it ever be - or is this intended for future in-call device changes? If so my hunch would be that it would be better to ask for both audio & video again, so we always get them as one stream, etc.

If it is necessary, could we factor out the duplicated device ID / resolution between the two?

Reply:
In my head, this would be also useful for adding a video track to an already existing stream that only has an audio track (turning an audio call into a video call). Replacing the whole stream is also an option but I like this one better. But I don't have much WebRTC experience, so my feeling might be wrong...

I can definitely factor out the duplicate stuff

Reply:
OK, yeah - my thinking there is that by requesting audio and video together, its clear that you want audio+video, ie. with the same base timestamp, whereas by asking for a video stream separately there's more scope for the provider to do something weird like decide to give you a video stream with timestamps that bear no resemblance to the audio stream (especially since they'll usually come from separate devices anyway).

Reply:
Ok, that makes sense, I'll rework this later